frontend/src/components/penreg/penrequest-batch/prb-student-details/InfoDialog.vue

```diff
@@ -55,25 +74,79 @@
         <PrimaryButton id="requestInfoDialogPostBtn" text="Post" :disabled="requestInfoDialogText===null" @click.native="$emit('updateInfoRequested', requestInfoDialogText)"></PrimaryButton>
       </v-card-actions>
     </v-card>
+    <ConfirmationDialog ref="confirmationDialog"></ConfirmationDialog>
   </v-dialog>
 </template>
 
 <script>
 import PrimaryButton from '../../../util/PrimaryButton';
+import { replaceMacro } from '../../../../utils/macro';
+import { mapState } from 'vuex';
+import ConfirmationDialog from '../../../util/ConfirmationDialog';
 export default {
   name: 'InfoDialog',
-  components: {PrimaryButton},
+  components: {
+    PrimaryButton,
+    ConfirmationDialog
+  },
+  props: {
+    text: {
+      type: String,
+      default: ''
+    }
+  },
   data() {
     return {
       requestInfoDialogOpen: false,
-      requestInfoDialogText: null,
-      messageDropDownItems: []
+      requestInfoDialogText: '',
+      macroAlert: false,
+      macroAlertMessage: 'Insufficient space remaining for this macro'
     };
   },
+  mounted() {
+    this.$store.dispatch('penRequestBatch/getMacros');
+    this.requestInfoDialogText = this.text;
+  },
+  computed: {
+    ...mapState('penRequestBatch', ['studentInfoMacros'])
+  },
   methods: {
-    closeRequestInfoDialog() {
-      this.requestInfoDialogOpen = false;
-      this.requestInfoDialogText = null;
+    async closeRequestInfoDialog() {
+      if(this.requestInfoDialogText !== this.text && !(!this.requestInfoDialogText && !this.text)) {
```

Comment:
Could we make the logic operation simpler? just (this.requestInfoDialogText !== this.text && (this.requestInfoDialogText || this.text))